python/runfiles/runfiles.py

```diff
@@ -153,14 +187,15 @@ def Rlocation(self, path, source_repo=None):
             #   which also should not be mapped.
             return self._strategy.RlocationChecked(path)
 
+        assert source_repo is not None
```

Comment:
consider adding an explanation to the `assert`. If this assert fails, having no message may require the user to read the code. Maybe something like the following could work.

```suggestion
        assert source_repo is not None, f"BUG: if the `source_repo` is None, we should never go past the `if` statement above"
```